### PR TITLE
Fix syntax of Typeclasses eauto := in reference manual.

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** documentation fix

### CODE REVIEW

File: doc/sphinx/addendum/type-classes.rst [hunk 1]
```diff
@@ -582,9 +582,9 @@ Typeclasses eauto `:=`
    + ``debug`` In debug mode, the trace of successfully applied tactics is
      printed.  This value can also be set with :flag:`Typeclasses Debug`.
 
-   + ``dfs, bfs`` This sets the search strategy to depth-first search (the
+   + ``(dfs)``, ``(bfs)`` This sets the search strategy to depth-first search (the
      default) or breadth-first search.  This value can also be set with
```
Comment (jfehrle) [inline doc/sphinx/addendum/type-classes.rst]: Maybe change "This value" -> "The search strategy".  IHMO, a "search strategy" is a setting that has a value--but is not itself a value.

File: doc/sphinx/addendum/type-classes.rst [hunk 2]
```diff
@@ -582,9 +582,9 @@ Typeclasses eauto `:=`
...
      :flag:`Typeclasses Iterative Deepening`.
 
-   + ``depth`` This sets the depth limit of the search.  This value can also be set with
-     :opt:`Typeclasses Depth`.
+   + :token:`num` This sets the depth limit of the search. This value
```
Comment (jfehrle) [inline doc/sphinx/addendum/type-classes.rst]: Similarly, "This value" -> "The depth limit"

**Review by jfehrle — APPROVED** [review]
2 minor wording suggestions.

**Comment by Zimmi48** [comment]
Thanks for the suggestions Jim! Updated.

**Comment by jfehrle** [comment]
Would you like me to merge this?
If so, do I just click "Merge pull request" button?

**Comment by Zimmi48** [comment]
> Would you like me to merge this?

Yes.

> If so, do I just click "Merge pull request" button?

No, you should use the merge script as documented in https://github.com/coq/coq/blob/master/dev/doc/MERGING.md. Is it the first time you use it?

**Comment by jfehrle** [comment]
Yes, first time for the merge script.

**Comment by Zimmi48** [comment]
Let me know if you encounter any issues. There are some dependencies (in principle they should be checked when running the script and the error message should be explicit) that you might have to install.

**Comment by jfehrle** [comment]
Ah, I hope this is really simple.  I've spent at least the last week just dealing with problems rather than making progress on my original goal :-(.

First issue:

```
$ git pull
From https://github.com/jfehrle/coq
 + 04e09be3da...fa35dd6e40 doc_toc    -> origin/doc_toc  (forced update)
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=<remote>/<branch> master
```

Is it sufficient to do a `git pull upstream master`, which is what I usually use, and perhaps a similar pull for this branch that I'll be merging?

**Comment by jfehrle** [comment]
OK, think I figured out that bit.

**Comment by Zimmi48** [comment]
So FTR, the merger needs to first checkout master before attempting the merge with the script.

@jfehrle If you encounter too many problems, we can also ask @cpitclaudel to take care of merging.

**Comment by jfehrle** [comment]
It would be a nice touch for the script to check for all the dependencies at once since I had to install both jz and gpg.  Each install requires logging out, a bit of overhead.  jz required renaming the binary from jz-win64.exe to jz.exe.  How do you recover if you lose your GPG private key?

The script uses some strange highlights.  I tried changing color settings but the only way I can see some of the info is to copy the text to an editor.  Also "nfo:" should be "info:"  Is this OK?

image

```
$ dev/tools/merge-pr.sh 9828
 nfo: title for PR 9828 is Fix syntax of Typeclasses eauto := in reference manual.
 nfo: PR 9828 targets branch coq:master
info: you are merging in master
info: remote for master is upstream
info: fetching from upstream the PR
Fetching upstream
remote: Enumerating objects: 23, done.
remote: Counting objects: 100% (19/19), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 11 (delta 7), reused 11 (delta 7), pack-reused 0
Unpacking objects: 100% (11/11), done.
From https://github.com/coq/coq
   0ff2e4b9f8..2ac275c0f3  master     -> upstream/master
remote: Enumerating objects: 6, done.
remote: Counting objects: 100% (6/6), done.
remote: Total 6 (delta 5), reused 6 (delta 5), pack-reused 0
Unpacking objects: 100% (6/6), done.
From https://github.com/coq/coq
 * branch                  refs/pull/9828/head -> FETCH_HEAD
info: commit for PR 9828 is 84f7aa2cdfbe6066c482805336e656b9de6d4762
 but you are merging in master :master
Continue anyway? [y/N]
```

**Comment by SkySkimmer** [comment]
I believe the color codes are handled by your terminal emulator, what are you using? It seems like yours eats some random text around colors.

**Comment by jfehrle** [comment]
> I believe the color codes are handled by your terminal emulator, what are you using? It seems like yours eats some random text around colors.

The usual Windows command window.  I get a better appearance from mintty but it also dropped an "i" in a different spot.  The shell script code appears correct.  I didn't have similar troubles when I was creating the diff feature.  Not worth diving deep into color highlighting now.

Should I say "Y" or "n" here?

**Comment by Zimmi48** [comment]
@jfehrle The display is not right, here is what it should look like if you encounter such an error message:

```
info: title for PR 9828 is Fix syntax of Typeclasses eauto := in reference manual. 
info: PR 9828 targets branch coq:master 
info: you are merging in test 
info: remote for test is upstream 
info: fetching from upstream the PR 
Récupération de upstream
Depuis github.com:coq/coq
 * branch                  refs/pull/9828/head -> FETCH_HEAD
info: commit for PR 9828 is 84f7aa2cdfbe6066c482805336e656b9de6d4762 
error: PR requests merge in coq:master but you are merging in test 
Continue anyway? [y/N]
```

So part of the error message was truncated and the leading `:master` should have appeared before.

To reproduce this error message, I created a test branch and set its upstream to upstream/master.

In your case though, it looks like you are trying to merge into the right branch so it looks like the check is failing for some reason.

You could try adding debug information like this:

```patch
--- a/dev/tools/merge-pr.sh
+++ b/dev/tools/merge-pr.sh
@@ -116,6 +116,9 @@ info "commit for PR $PR is ${BLUE}$COMMIT"
 
 # Sanity check: merge to a different branch
 
+echo "$BASE_BRANCH"
+echo "coq:$CURRENT_LOCAL_BRANCH"
+
 if [ "$BASE_BRANCH" != "coq:$CURRENT_LOCAL_BRANCH" ]; then
   error "PR requests merge in ${BLUE}$BASE_BRANCH${RESET} but you are merging in ${BLUE}$CURRENT_LOCAL_BRANCH"
   ask_confirmation
```

Note that @MSoegtropIMC often uses the merge script on Windows, so at least in his setting, it works. But he may be using the cygwin console, while you are using IIUC the Windows cmd window...

**Comment by jfehrle** [comment]
It appears that some of the shell variables include return characters with a trailing space, so characters following the return overwrite what was previously printed.  See the first character on the last line below.  The comparison fails because one string is "coq:master" and the other is "coq:master\r".  Perhaps the JSON fetch is doing this.

If I run from the bash shell, the output formatting is fine but git doesn't know my gpg key.  When I installed gpg, it ran a Kleopatra utility to create a key but it appears gpg doesn't know about it.  I guess these are the instructions I should follow: https://help.github.com/en/articles/generating-a-new-gpg-key and create another key?

```
$ dev/tools/merge-pr.sh 9828 >out

$ od -bc out
0000000 033 133 063 062 155 151 156 146 157 072 033 133 060 155 040 164
        033   [   3   2   m   i   n   f   o   : 033   [   0   m       t
0000020 151 164 154 145 040 146 157 162 040 120 122 040 071 070 062 070
          i   t   l   e       f   o   r       P   R       9   8   2   8
0000040 040 151 163 040 033 133 063 064 155 106 151 170 040 163 171 156
              i   s     033   [   3   4   m   F   i   x       s   y   n
0000060 164 141 170 040 157 146 040 124 171 160 145 143 154 141 163 163
          t   a   x       o   f       T   y   p   e   c   l   a   s   s
0000100 145 163 040 145 141 165 164 157 040 072 075 040 151 156 040 162
          e   s       e   a   u   t   o       :   =       i   n       r
0000120 145 146 145 162 145 156 143 145 040 155 141 156 165 141 154 056
          e   f   e   r   e   n   c   e       m   a   n   u   a   l   .
0000140 015 040 033 133 060 155 012 033 133 063 062 155 151 156 146 157
         \r     033   [   0   m  \n 033   [   3   2   m   i   n   f   o
```

```
$ dev/tools/merge-pr.sh 9828
info: title for PR 9828 is Fix syntax of Typeclasses eauto := in reference manua                  l.
info: PR 9828 targets branch coq:master
info: you are merging in master
info: remote for master is upstream
info: fetching from upstream the PR
Fetching upstream
From https://github.com/coq/coq
 * branch                  refs/pull/9828/head -> FETCH_HEAD
info: commit for PR 9828 is 84f7aa2cdfbe6066c482805336e656b9de6d4762
warning: git config user.signingkey is empty
warning: gpg will guess a key out of your git config user.* data
info: Fetching review data
info: merging
fatal: You have not concluded your merge (MERGE_HEAD exists).
Please, commit your changes before you merge.

$ gpg --list-secret-keys --keyid-format LONG
gpg: directory `/c/Users/Jim.DESKTOP-MMPNVGH/.gnupg' created
gpg: new configuration file `/c/Users/Jim.DESKTOP-MMPNVGH/.gnupg/gpg.conf' creat                  ed
gpg: WARNING: options in `/c/Users/Jim.DESKTOP-MMPNVGH/.gnupg/gpg.conf' are not                   yet active during this run
gpg: keyring `/c/Users/Jim.DESKTOP-MMPNVGH/.gnupg/secring.gpg' created
gpg: keyring `/c/Users/Jim.DESKTOP-MMPNVGH/.gnupg/pubring.gpg' created
gpg: /c/Users/Jim.DESKTOP-MMPNVGH/.gnupg/trustdb.gpg: trustdb created

$ gpg --list-secret-keys --keyid-format LONG

$
```

**Comment by jfehrle** [comment]
I've been regularly using the cygwin shell which I start with a Windows batch file that was part of the OCaml distribution I originally installed (directly, not through OPAM).  I configured my gpg keys in this shell.  As I noted above, that has return characters in the data so the script doesn't work properly.  Home directory for this shell is `C:\OCaml64\home\Jim`.

I also have a Git Bash installed that doesn't have the return character problem.  I really don't use this one. It seems somewhat crippled (e.g., no 'man' command).  It has a different home directory `C:\Users\Jim.DESKTOP-MMPNVGH` so it doesn't see the gpg configuration.  Copying the .gnupg directory from one home to another doesn't fix it.

If I modify the merge script to strip out `\r` from the jq output, the script appears to work correctly in the cygwin shell.

**Comment by jfehrle** [comment]
Did this work correctly?

I noticed my `upstream` is `https://github.com/coq/coq (push)` rather than `git@github.com:coq/coq.git`.  Does that matter?

**Comment by Zimmi48** [comment]
Thanks Jim! It worked correctly. The upstream URL is fine. I'd advise against creating multiple gpg keys. You generally want to have only one of these (and let GitHub know about the public key, so that it doesn't show your signature as "unverified" https://github.com/coq/coq/commits/21e9d207a8d8a0bee74c000f02a4e2819c53fae5).

**Comment by silene** [comment]
This depends on #8911, which is not flagged as 8.9 material. So, not backporting.

**Pull request merged**